lib/Doctrine/ORM/Tools/Export/Driver/XmlExporter.php

```diff
@@ -451,4 +454,61 @@ private function _asXml($simpleXml)
 
         return $dom->saveXML();
     }
+
+    /**
+     * @param ClassMetadataInfo $metadata
+     * @param SimpleXMLElement  $root
+     */
+    private function processEntityListeners(ClassMetadataInfo $metadata, SimpleXMLElement $root)
+    {
+        if (0 !== \count($metadata->entityListeners)) {
+            $entityListenersXml = $root->addChild('entity-listeners');
+            $entityListenersXmlMap = [];
+
+            $this->generateEntityListenerXml($metadata, $entityListenersXmlMap, $entityListenersXml);
+        }
+    }
+
+
+    /**
+     * @param ClassMetadataInfo $metadata
+     * @param array             $entityListenersXmlMap
+     * @param SimpleXMLElement  $entityListenersXml
+     */
+    private function generateEntityListenerXml(ClassMetadataInfo $metadata, array $entityListenersXmlMap, SimpleXMLElement $entityListenersXml)
+    {
+        foreach ($metadata->entityListeners as $event => $entityListenerConfig) {
+            foreach ($entityListenerConfig as $entityListener) {
+                $entityListenerXml = $this->addClassToMapIfExists(
+                    $entityListenersXmlMap,
+                    $entityListener,
+                    $entityListenersXml
+                );
+
+                $entityListenerCallbackXml = $entityListenerXml->addChild('lifecycle-callback');
+                $entityListenerCallbackXml->addAttribute('type', $event);
+                $entityListenerCallbackXml->addAttribute('method', $entityListener['method']);
+            }
+        }
+    }
+
+    /**
+     * @param array            $entityListenersXmlMap
+     * @param array            $entityListener
+     * @param SimpleXMLElement $entityListenersXml
+     *
+     * @return SimpleXMLElement
+     */
+    private function addClassToMapIfExists(array $entityListenersXmlMap, array $entityListener, SimpleXMLElement $entityListenersXml)
+    {
+        if (!isset($entityListenersXmlMap[$entityListener['class']])) {
+            $entityListenerXml = $entityListenersXml->addChild('entity-listener');
+            $entityListenerXml->addAttribute('class', $entityListener['class']);
+            $entityListenersXmlMap[$entityListener['class']] = $entityListenerXml;
+        } else {
+            $entityListenerXml = $entityListenersXmlMap[$entityListener['class']];
```

Comment:
Use an early return. Swap the negation on the `if` to get rid of the `else` overall